scripts/west_commands/runners/bossac.py

```diff
@@ -96,27 +83,65 @@ def is_extended_samba_protocol(self):
                 return True
         return False
 
-    def get_offset(self, supports_offset):
+    def is_partition_enabled(self):
+        build_conf = BuildConfiguration(self.cfg.build_dir)
+
+        if 'CONFIG_USE_DT_CODE_PARTITION' not in build_conf:
+            return False
+
+        return True
+
+    def get_chosen_code_partition_node(self):
+        # Get the EDT Node corresponding to the zephyr,code-partition
+	# chosen DT node
+
+        # Ensure the build directory has a compiled DTS file
+        # where we expect it to be.
+        b = pathlib.Path(self.cfg.build_dir)
+        edt_pickle = b / 'zephyr' / 'edt.pickle'
+        if not edt_pickle.is_file():
+            error_msg = "can't load devicetree; expected to find:" \
+	                + str(edt_pickle)
+
+            raise NotImplementedError(error_msg)
+
+        # Load the devicetree.
+        with open(edt_pickle, 'rb') as f:
+            edt = pickle.load(f)
+
+        return edt.chosen_node('zephyr,code-partition')
+
+    def get_board_name(self):
+        build_conf = BuildConfiguration(self.cfg.build_dir)
+
+        if 'CONFIG_BOARD' not in build_conf:
+            return '<board>'
```

Comment:
It is not a error handler. It is used to print the board name at an error handler to user and I simple added a fallback.

Reply:
> It is not a error handler. It is used to print the board name at an error handler to user and I simple added a fallback.

I understand the code. What I'm saying is that if CONFIG_BOARD is missing, the build configuration is **really** messed up. That's a much bigger problem than CONFIG_USE_DT_CODE_PARTITION not being set.

The error message is therefore misleading, because build without CONFIG_BOARD set at all is going to fail for many other bigger reasons. So I'm wondering if the error message could be improved.

But like I said, not a blocker.